lib/versioning/VersionID.ts

```diff
@@ -122,6 +143,20 @@ export function generateVersionId(info: string, replicationGroupId: string): str
     // replication group ID, like PARIS; will be trimmed if exceed LENGTH_RG
     const repGroupId = padRight(replicationGroupId, TEMPLATE_RG);
 
+    let otherInfo = '';
+    let instanceIdPadded = '';
+    let formatSuffix = '';
+
+    if (S3_VERSION_ID_ENCODING_TYPE === 'hex' || !S3_VERSION_ID_ENCODING_TYPE) {
```

Comment:
I would reverse the check order: if the env var is not set, no need to check for the `hex` value